qulacsvis/models/circuit.py

```diff
@@ -0,0 +1,95 @@
+import dataclasses
+from collections import deque
+from collections.abc import Sequence
+from itertools import chain
+from typing import Deque, List, Optional, Tuple
+
+import dataclasses_json
+
+
+@dataclasses_json.dataclass_json
+@dataclasses.dataclass
+class ControlQubitInfo:
+    index: int
+    control_value: int
+
+
+@dataclasses_json.dataclass_json
+@dataclasses.dataclass
+class GateData:
+    name: str
+    target_bits: List[int] = dataclasses.field(default_factory=list)
+    control_bit_infos: List[ControlQubitInfo] = dataclasses.field(default_factory=list)
+
+    @property
+    def indices(self) -> Tuple[int, ...]:
+        return tuple(chain(self.target_bits, (c.index for c in self.control_bit_infos)))
+
+    @property
+    def max_index(self) -> int:
+        return max(self.indices)
+
+    @property
+    def min_index(self) -> int:
+        return min(self.indices)
+
+
+GateDataSeq = Sequence[Sequence[GateData]]
+
+
+@dataclasses_json.dataclass_json
+@dataclasses.dataclass
+class CircuitData:
+    qubit_count: int
+    layer_count: int
+    gates: GateDataSeq
+
+    @staticmethod
+    def from_gate_sequence(
+        gates: Sequence[GateData], qubit_count: Optional[int] = None
+    ) -> "CircuitData":
```

Comment:
`from_gate_sequence`にdocstringがあると嬉しいです！

```suggestion
    ) -> "CircuitData":
    """
    Construct a CircuitData model from a list of gates.
    
    Parameters
    ----------
    gates : Sequence[GateData]
        List of gates
    qubit_count : int optional
        Number of qubits this quantum circuit has
        
    Returns
    -------
    CircuitData : CircuitData
        Constructed CircuitData model
    """
```